pkg/kube/inject/webhook.go

```diff
@@ -970,17 +970,23 @@ func findProxyUGID(pod corev1.Pod) (proxyUID, proxyGID *int64) {
 		if proxyUID == nil {
 			if pod.Spec.SecurityContext != nil && pod.Spec.SecurityContext.RunAsUser != nil {
 				proxyUID = pointer.Int64(*pod.Spec.SecurityContext.RunAsUser + 1)
-				// valid GID for fsGroup defaults to first int in UID range in OCP's restricted SCC
-				proxyGID = pod.Spec.SecurityContext.RunAsUser
+				proxyGID = pointer.Int64(*proxyUID)
+				if pod.Spec.SecurityContext.RunAsGroup != nil {
+					proxyGID = pointer.Int64(*pod.Spec.SecurityContext.RunAsGroup + 1)
+				}
 			}
 			for _, c := range pod.Spec.Containers {
 				if c.Name != ProxyContainerName && c.SecurityContext != nil && c.SecurityContext.RunAsUser != nil {
 					uid := *c.SecurityContext.RunAsUser + 1
 					if proxyUID == nil || uid > *proxyUID {
 						proxyUID = &uid
 					}
-					if proxyGID == nil {
-						proxyGID = c.SecurityContext.RunAsUser
+					gid := uid
+					if c.SecurityContext.RunAsGroup != nil {
+						gid = *c.SecurityContext.RunAsGroup + 1
+					}
+					if proxyGID == nil || gid > *proxyGID {
```

Comment:
why do we check for `gid > *proxyGID`? Shouldn't the container's securityContext always take precedence, even if the int is lower?

Reply:
I am not the original author of this condition, so I do not know... But I would also expect that container SC takes precedence over pod SC.

Reply:
All this code needs to do is find an UID and a GID that isn't used by any of the containers (i.e. it isn't specified in the pod's nor the container's `securityContext`).

But what the code effectively does is find the max UID/GID used by the containers and then adds a `+ 1`  to that value. Although this isn't wrong, it's not the best, as it wouldn't work if one of the containers uses the last value of the uid/gid range available in the pod's namespace.

So, ideally, we should fix this and return the first unused uid/gid we find. But I guess it's also okay if we leave the current behavior. It looks correct, apart from the corner case described above.

Reply:
> So, ideally, we should fix this and return the first unused uid/gid we find. But I guess it's also okay if we leave the current behavior. It looks correct, apart from the corner case described above.

@luksa do you suggest that my fix is incorrect? Now, this function finds first unused UID/GID, while previously it was only finding the first unused UID, but not GID.

Reply:
No, I was just commenting on the `gid > *proxyGID` (which came from `uid > *proxyUID`). 

Your fix looks correct, but we should probably fix both uid and gid so that take the first unused UID/GID instead of taking `max(uid across all containers)+1`.

Reply:
Ok, thanks for clarifying.

Reply:
we can consider a fix like that for upstream, but let's not overcomplicate the bugfix